src/spandrel/__helpers/model_descriptor.py

```diff
@@ -124,17 +124,52 @@ def __init__(
 
         self.model.load_state_dict(state_dict)  # type: ignore
 
-    def to(self, device: torch.device):
-        self.model.to(device)
+    def to(
+        self,
+        device: int | torch.device | None = torch.device("cpu"),
+        dtype: torch.dtype | str | None = None,
+        non_blocking: bool = False,
+    ):
+        self.model.to(device, dtype, non_blocking)
+        return self
+
+    def half(self):
+        self.model.half()
+        return self
+
+    def float(self):
+        self.model.float()
         return self
 
+    def cuda(self):
+        self.model.cuda()
+        return self
+
+    def cpu(self):
+        self.model.cpu()
+        return self
+
+    def eval(self):
+        self.model.eval()
+        return self
+
+    def __call__(self, *args, **kwargs):  # noqa: ANN002
+        return self.model(*args, **kwargs)
+
 
 class SRModelDescriptor(ModelBase[T], Generic[T]):
-    pass
+    def __call__(self, image: torch.Tensor) -> torch.Tensor:
+        """
+        Upscale the given image by the scale factor of the model.
+        """
+        return self.model(image)
 
 
 class FaceSRModelDescriptor(ModelBase[T], Generic[T]):
-    pass
+    def __call__(
+        self, image: torch.Tensor, return_rgb: bool = False, weight: float = 0.5
```

Comment:
What is `return_rgb` and `weight`? 

`outputChannels` decides the number of output channel, so what is `return_rgb`?

Reply:
It's used by the model code. https://github.com/chaiNNer-org/spandrel/blob/11aae676a9ee4dab7091424ef0ba7d9f7344f730/src/spandrel/architectures/GFPGAN/arch/gfpganv1_clean_arch.py#L302

Reply:
Not really sure what return RGB is, but chaiNNer passes that into the method

Reply:
Joey, an API is supposed to abstract over implementation details. The API for the call function you made is that it returns *one* image, so how is `return_rgb` relevant?